website/docs/fun/counting_v2.md

```diff
@@ -0,0 +1,82 @@
+---
+title: Counting System
+---
+
+This command implements a counting game with detailed statistics and a leaderboard.
+
+A unique feature of this command is that, besides decimal numbers, it also accepts roman numerals and mathematical expressions as valid inputs. So if the next number is 2, `II` and `2*sin(pi/2)` would be considered correct as well.
+
+:::note
+
+Unlike some other counting systems, this command does not delete invalid numbers to allow counting and other conversation to take place concurrently. Messages that include text which do not correspond to valid numbers will, however, trigger a warning message.
```

Comment:
I think your first suggestion reads somewhat incorrectly; it reads to me that if I were to throw a 1 somewhere in the message, it would be valid, even if the expression sent is `invalid1exp`. However, this does make me notice the usage of the word "numbers" here, which may make the user question if terms other than decimal numbers are accepted. I think it'd be better to use "terms" in replacement.

As for the question and suggestion on first line expressions, yes, only the first line of the message is parsed for an expression. I believe the previous paragraph was the result of a [previous discussion](https://github.com/yagpdb-cc/yagpdb-cc/pull/334#discussion_r1685176559). It could use some expanding to make it known only the first line is parsed and the rest is ignored.

I will not make any changes to this now so it can be discussed.

Reply:
```suggestion
Unlike some other counting systems, this command does not delete invalid expressions to allow counting and other conversation to take place concurrently. Expressions which include unaccepted terms or evaluate to an invalid expression will, however, trigger a warning message.
```

Reply:
> I think your first suggestion reads somewhat incorrectly; it reads to me that if I were to throw a 1 somewhere in the message, it would be valid, even if the expression sent is invalid1exp

That was not the point of the first suggestion. The point was to simplify this sentence, the second suggestion expanded on that by asking to note down the behaviour with regards to the first line of a message.


> I think it'd be better to use "terms" in replacement.

No, you already defined what counts as a valid number for the purposes of this custom command, why introduce another term? There's "number" and "valid number", I don't see how replacing that with "expression" makes things clearer.

Even still, your suggestion doesn't make this any clearer anyway, as you're not stating that this "expression" has to occur on the first line.

Reply:
I understand the point of the first suggestion was for simplification. However, I am saying that from the rewrite, it does not read the same.

I think using "number" in the docs will not influence the users to use the command to it's potential, which is fine if they don't want to use roman numerals and/or math. But I think it scopes in on decimal numbers as the message to send, and not the result of an expression.

I did not include the second line as I wasn't confident in including it, and can be committed separately after this discussion. Overall, yes, I agree it should be noted only the first line is to be parsed.

Reply:
Sounds like we came to an agreement. Add these notes and then we should be good to go.

Reply:
I've updated the note to be more direct in my opinion and added the first line suggestion.